### DISCUSSION TITLE
How to track resources generates by a cluster scope resource?

### DISCUSSION THREAD

**Discussion by alexisbel1** [argoproj/argo-cd#8047] [Q&A]
I use Kubevela (Open Application Model specs) and ArgoCD to manage and deploy applications on top of Kubernetes. Kubevela uses an Application  CRD to generate deployments, services, ingress etc... All generated resources are managed by a `ResourceTracker` CRD, which is a cluster scope resource.

What surprise me is that generated resources seems to be tracked by the argocd controller (app.kubernetes.io/instance: annotation has been injected, but i don't know how) and appears (as expected) out of sync (prune required). I use argocd v2.2.1:

![image](https://user-images.githubusercontent.com/25341574/147591195-10f0c683-dc92-445f-a176-0c02e57ab1ab.png)

I have 2 questions:

- How deployment, service, ingress resources have been tracked by argoCD while only the Kubevela application CRD is defined in Git and there is no "owner" for those resources managed by a cluster scope resource?
- What is the best way to track resources generated by a cluster scope resource?

Thanks a lot!

**Comment by pasha-codefresh** [comment]
Hi @alexisbel1 . I think it is a similar issue with this. https://github.com/argoproj/argo-cd/issues/3603

You can see a proposal to solve this that has already been implemented in argocd 2.2

https://github.com/argoproj/argo-cd/blob/master/docs/proposals/application-name-identifier.md

In the motivation section, you can see,

Popular off-the-shelf Helm charts often add the app.kubernetes.io/instance label to the generated resource manifests. This label confuses Argo CD and makes it think the resource is managed by the application.

To solve that you need to go to this doc https://argo-cd.readthedocs.io/en/stable/user-guide/resource_tracking/ and define tracking method with annotations 

Let me know if you need some help with that
